src/sage/misc/c3.pyx

```diff
@@ -192,9 +192,10 @@ cpdef list C3_algorithm(object start, str bases, str attribute, bint proper):
     cdef object O, X
     cdef list tails = [getattr(obj, attribute) for obj in args]
     tails.append(args)
-    tails              = [list(reversed(tail))                   for tail in tails]
-    cdef list heads    = [tail.pop()                             for tail in tails]
-    cdef list tailsets = [set([<size_t><void *>O for O in tail]) for tail in tails]
+    tails = [list(reversed(tail)) for tail in tails]
+    cdef list heads = [tail.pop() for tail in tails]
+    cdef list tailsets = [set([<size_t><void *>O for O in tail])
```

Comment:
you can remove the `[...]` in `set(...)`, so `cdef list tailsets = [set(<size_t><void *>O for O in tail)`

Reply:
note also that in `src/sage/misc/c3_controlled.pyx` line 620, the `<size_t><void *>` is commented. Is it useful ?